src/pojo-interfaces.ts

```diff
@@ -0,0 +1,418 @@
+}
+
+type QuadParseOptions = Partial<{
+  otherQuads: RdfJs.Quad[];
+}>;
+
+function addRdfJsQuadToDataset(
+  dataset: Dataset,
+  quad: RdfJs.Quad,
+  quadParseOptions: QuadParseOptions = {}
+): Dataset {
+  const supportedGraphTypes: Array<typeof quad.graph.termType> = [
+    "NamedNode",
+    "DefaultGraph",
+  ];
+  if (!supportedGraphTypes.includes(quad.graph.termType)) {
+    throw new Error(
+      `Cannot parse Quads with nodes of type [${quad.graph.termType}] as their Graph node.`
+    );
+  }
+  const graphId =
+    quad.graph.termType === "DefaultGraph" ? "default" : quad.graph.value;
+
+  const graph: Graph = dataset.graphs[graphId] ?? {};
+  return freeze({
+    ...dataset,
+    graphs: freeze({
+      ...dataset.graphs,
+      [graphId]: addRdfJsQuadToGraph(graph, quad, quadParseOptions),
+    }),
+  });
+}
+
+function addRdfJsQuadToGraph(
+  graph: Graph,
+  quad: RdfJs.Quad,
+  quadParseOptions: QuadParseOptions = {}
+): Graph {
+  const supportedSubjectTypes: Array<typeof quad.subject.termType> = [
+    "NamedNode",
+  ];
+  if (!supportedSubjectTypes.includes(quad.subject.termType)) {
+    throw new Error(
+      `Cannot parse Quads with nodes of type [${quad.subject.termType}] as their Subject node.`
+    );
+  }
+
+  const subjectIri = quad.subject.value;
+
+  const subject: Subject = graph[subjectIri] ?? {
+    type: "Subject",
+    url: subjectIri,
+    predicates: {},
+  };
+  return freeze({
+    ...graph,
+    [subjectIri]: addRdfJsQuadToSubject(subject, quad, quadParseOptions),
+  });
+}
+
+function addRdfJsQuadToSubject(
+  subject: Subject,
+  quad: RdfJs.Quad,
+  quadParseOptions: QuadParseOptions = {}
+): Subject {
+  return freeze({
+    ...subject,
+    predicates: addRdfJsQuadToPredicates(
+      subject.predicates,
+      quad,
+      quadParseOptions
+    ),
+  });
+}
+
+function addRdfJsQuadToPredicates(
+  predicates: Predicates,
+  quad: RdfJs.Quad,
+  quadParseOptions: QuadParseOptions = {}
+): Predicates {
+  const supportedPredicateTypes: Array<typeof quad.predicate.termType> = [
+    "NamedNode",
+  ];
+  if (!supportedPredicateTypes.includes(quad.predicate.termType)) {
+    throw new Error(
+      `Cannot parse Quads with nodes of type [${quad.predicate.termType}] as their Predicate node.`
+    );
+  }
+  const predicateIri = quad.predicate.value;
+  const objects = predicates[predicateIri] ?? {};
+  return freeze({
+    ...predicates,
+    [predicateIri]: addRdfJsQuadToObjects(objects, quad, quadParseOptions),
+  });
+}
+
+function addRdfJsQuadToObjects(
+  objects: Objects,
+  quad: RdfJs.Quad,
+  quadParseOptions: QuadParseOptions = {}
+): Objects {
+  if (quad.object.termType === "NamedNode") {
+    const namedNodes = freeze([
+      ...(objects.namedNodes ?? []),
+      quad.object.value,
+    ]);
+    return freeze({
+      ...objects,
+      namedNodes: namedNodes,
+    });
+  }
+
+  if (quad.object.termType === "Literal") {
+    if (quad.object.datatype.value === xmlSchemaTypes.langString) {
```

Comment:
This is just a small thing, but we really should fix this ASAP (as it's 'infecting' this code too now) - i.e., `langString` is not an XML Schema type, it's defined in the [RDF vocab](https://www.w3.org/1999/02/22-rdf-syntax-ns#).

Reply:
I won't do it in this PR since it's not intended to be merged, but I wouldn't worry too much about it "infecting": it's internal-only and can be renamed to e.g. `knownDataTypeIriStringAll` (or whatever name you prefer) in one go in your IDE.

Reply:
Yeah, I wasn't trying to suggest fixing it as part of this PR, I just meant fixed in the main codebase. As you say, it's an easy fix - maybe we have ticket already, but I couldn't find one, so I created one (internal ID: SDK-1911).